java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -395,14 +395,26 @@ class DatabaseImpl(
                 ) {
                     val rawSingletons = mutableMapOf<FieldName, Referencable?>()
                     val rawCollections = mutableMapOf<FieldName, Set<Referencable>>()
+                    val inlineStorageKeyId = it.getLong(3)
+                    val entityId = db.rawQuery(
+                        """
+                            SELECT
+                                entity_id
+                            FROM entities
+                            WHERE storage_key_id = ?
+                        """.trimIndent(),
+                        arrayOf(inlineStorageKeyId.toString())
+                    ).forSingleResult {
+                        it.getString(0)
+                    }
```

Comment:
I think this extra query could be merged into the getEntityFields query to avoid an extra round-trip.. but that optimization could wait for a later PR.

Reply:
I'll file an issue to track.